pkg/controller/compliancescan/compliancescan_controller.go

```diff
@@ -144,6 +145,92 @@ func (r *ReconcileComplianceScan) Reconcile(request reconcile.Request) (reconcil
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileComplianceScan) handleRootCASecret() error {
+	secret := &corev1.Secret{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: "compliance-operator-root-ca", Namespace: "openshift-compliance"}, secret)
+	if err == nil {
+		// Already exists, so return.
+		return nil
+	}
+	if !errors.IsNotFound(err) {
+		return err
+	}
+
+	// Create the CA.
+	cert, key, err := utils.ComplianceOperatorRootCA("compliance-operator-root-ca", 100)
+	if err != nil {
+		return err
+	}
+
+	secret = &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "compliance-operator-root-ca",
+			Namespace: "openshift-compliance",
+		},
+		Data: map[string][]byte{
+			"ca.crt": cert,
+			"ca.key": key,
+		},
+	}
```

Comment:
Do you think it would make sense to wrap creating the Secret in a function so that we don't have to leak the key and cert parameters to the main controller code?

Reply:
done, I moved these to helper functions in a different file.